### PR TITLE
basic auth, solved some issues

### LINKED ISSUE #57: Extract function for user check
https://github.com/LialinMaxim/SSkey/blob/01ef73fe56e51db24dd802e95a7334fda7c8b385/src/app/resources.py#L123

User check appears more than once.

### PR DESCRIPTION
**Pull request by vromanuk**
added basic authentication
resolved #57, #37 
added pycodestyle to travis that validates code to pep8, also added coverage that failed if test coverage is under 20%

### CODE REVIEW

File: src/app/models.py [hunk 1]
```diff
@@ -123,6 +126,31 @@ def __init__(self, login, password, user_id, url, title, comment):
         self.comment = comment
 
 
+class FilterUserBy:
```
Comment (ArturM94) [inline src/app/models.py]: May be better if class name will be FilterUser, and function names will be by_email(), by_username() and by_id().

Comment (vromanuk) [reply src/app/models.py]: no, cause as I remember function name must be verbs

File: src/app/__init__.py [hunk 2]
```diff
@@ -1,17 +1,12 @@
...
-api = Api(app, version='0.1.1', title='SSkey', description='A simple application to safe yours passwords',)
+api = Api(app, version='0.1.1', title='SSkey', description='A simple application to safe yours passwords', )
 
-# app.config["SECRET_KEY"] = "5791628bb0b13ce0c676dfde280ba245"
-# db = ...
-
-# from src.app.errors import errors
-
-# app.register_blueprint(errors)
+app.config["SECRET_KEY"] = "5791628bb0b13ce0c676dfde280ba245"
```
Comment (ArturM94) [inline src/app/__init__.py]: We have default SECRET_KEY value in config.py (class DevelopmentConfig)

File: README.md [hunk 3]
```diff
@@ -103,7 +104,7 @@ docker-compose run -p 5000:5000 web python manage.py
 Standalone unit tests run with:
 
 ```shell
-pytest
+python -m pytest tests/
```
Comment (ArturM94) [inline README.md]: Are you sure that we can run a unit tests by this command?

**Pull request closed without merging**